shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/HomeViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package co.nimblehq.avishek.phong.kmmic.presentation.module
+
+import co.nimblehq.avishek.phong.kmmic.domain.model.Survey
+import co.nimblehq.avishek.phong.kmmic.domain.usecase.GetSurveysUseCase
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.update
+
+data class HomeViewState(
+    val isLoading: Boolean
+) {
+    constructor() : this (true)
+}
+
+class HomeViewModel(
+    private val getSurveysUseCase: GetSurveysUseCase
+) : BaseViewModel() {
+
+    private val mutableViewState: MutableStateFlow<HomeViewState> =
+        MutableStateFlow(HomeViewState())
+
+    val viewState: StateFlow<HomeViewState> = mutableViewState
```

Comment:
We can use the underscore prefix for the backing properties, a common practice in the Kotlin world.
```suggestion
    private val _viewState: MutableStateFlow<HomeViewState> =
        MutableStateFlow(HomeViewState())
    val viewState: StateFlow<HomeViewState> = mutableViewState
```